lib/ansible/modules/windows/win_psmodule.ps1

```diff
@@ -1,186 +1,380 @@
 #!powershell
 
+# Copyright: (c) 2018, Wojciech Sciesinski <wojciech[at]sciesinski[dot]net>
 # Copyright: (c) 2017, Daniele Lazzari <lazzari@mailup.com>
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
 #Requires -Module Ansible.ModuleUtils.Legacy
 
-# win_psmodule (Powershell modules Additions/Removal)
+# win_psmodule (Windows PowerShell modules Additions/Removals/Updates)
 
-$params = Parse-Args $args -supports_check_mode $true
+$params = Parse-Args -arguments $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
 
 $name = Get-AnsibleParam -obj $params -name "name" -type "str" -failifempty $true
+$required_version = Get-AnsibleParam -obj $params -name "required_version" -type "str"
+$minimum_version = Get-AnsibleParam -obj $params -name "minimum_version" -type "str"
+$maximum_version = Get-AnsibleParam -obj $params -name "maximum_version" -type "str"
 $repo = Get-AnsibleParam -obj $params -name "repository" -type "str"
-$url = Get-AnsibleParam -obj $params -name "url" -type "str"
-$state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "present" -validateset "present", "absent"
+$state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "present" -validateset "present", "absent", "latest"
 $allow_clobber = Get-AnsibleParam -obj $params -name "allow_clobber" -type "bool" -default $false
-$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -default $false
+$skip_publisher_check = Get-AnsibleParam -obj $params -name "skip_publisher_check" -type "bool" -default $false
+$allow_prerelease = Get-AnsibleParam -obj $params -name "allow_prerelease" -type "bool" -default $false
 
-$result = @{"changed" = $false
-            "output" = ""
-            "nuget_changed" = $false
-            "repository_changed" = $false}
+$result = @{changed = $false}
```

Comment:
When I saying to remove the *_changed variables I was only talking about the ones that was added in this PR. Unfortunately we are stuck with the existing ones until we go through a deprecation cycle which is 4 releases. If we do want to deprecate `nuget_changed`, `repository_changed`, and `output` we should do that in a separate PR.